frontend/public/module/k8s/types.ts

```diff
@@ -0,0 +1,948 @@
+  initialDelaySeconds?: number;
+  timeoutSeconds?: number;
+  periodSeconds?: number;
+  successThreshold?: number;
+  failureThreshold?: number;
+} & Handler;
+
+export type ContainerLifecycleStage = 'postStart' | 'preStop';
+
+export type ContainerLifecycle = {
+  postStart?: Handler;
+  preStop?: Handler;
+};
+
+export type ResourceList = {
+  [resourceName: string]: string;
+};
+
+export type EnvVarSource = {
+  fieldRef?: {
+    apiVersion?: string;
+    fieldPath: string;
+  };
+  resourceFieldRef?: {
+    resource: string;
+    containerName?: string;
+    divisor?: string;
+  };
+  configMapKeyRef?: {
+    key: string;
+    name: string;
+  };
+  secretKeyRef?: {
+    key: string;
+    name: string;
+  };
+  configMapRef?: {
+    key?: string;
+    name: string;
+  };
+  secretRef?: {
+    key?: string;
+    name: string;
+  };
+  configMapSecretRef?: {
+    key?: string;
+    name: string;
+  };
+  serviceAccountRef?: {
+    key?: string;
+    name: string;
+  };
+};
+
+export type EnvVar = {
+  name: string;
+  value?: string;
+  valueFrom?: EnvVarSource;
+};
+
+export type ContainerPort = {
+  name?: string;
+  containerPort: number;
+  protocol: string;
+};
+
+export enum ImagePullPolicy {
+  Always = 'Always',
+  Never = 'Never',
+  IfNotPresent = 'IfNotPresent',
+}
+
+export type NodeAffinity = {
+  preferredDuringSchedulingIgnoredDuringExecution?: {
+    preference: Selector;
+    weight: number;
+  }[];
+  requiredDuringSchedulingIgnoredDuringExecution?: {
+    nodeSelectorTerms: Selector[];
+  };
+};
+
+export type PodAffinity = {
+  preferredDuringSchedulingIgnoredDuringExecution: {
+    podAffinityTerm: {
+      labelSelector?: Selector;
+      namespaces?: string[];
+      topologyKey: string;
+    };
+    weight?: number;
+  }[];
+  requiredDuringSchedulingIgnoredDuringExecution: {
+    labelSelector?: Selector;
+    namespaces?: string[];
+    topologyKey: string;
+  }[];
+};
+
+export type ContainerSpec = {
+  name: string;
+  volumeMounts?: VolumeMount[];
+  env?: EnvVar[];
+  livenessProbe?: ContainerProbe;
+  readinessProbe?: ContainerProbe;
+  lifecycle?: ContainerLifecycle;
+  resources?: {
+    limits?: ResourceList;
+    requested?: ResourceList;
+  };
+  ports?: ContainerPort[];
+  imagePullPolicy?: ImagePullPolicy;
+  [key: string]: any;
+};
+
+export type Volume = {
+  name: string;
+  [key: string]: any;
+};
+
+export type PodSpec = {
+  volumes?: Volume[];
+  initContainers?: ContainerSpec[];
+  containers: ContainerSpec[];
+  restartPolicy?: 'Always' | 'OnFailure' | 'Never';
+  terminationGracePeriodSeconds?: number;
+  activeDeadlineSeconds?: number;
+  nodeSelector?: any;
+  serviceAccountName?: string;
+  priorityClassName?: string;
+  tolerations?: Toleration[];
+  nodeName?: string;
+  hostname?: string;
+  [key: string]: any;
+};
+
+// https://github.com/kubernetes/api/blob/release-1.16/core/v1/types.go#L2411-L2432
+type PodPhase = 'Pending' | 'Running' | 'Succeeded' | 'Failed' | 'Unknown';
```

Comment:
Note, this is no longer exported as it would clash with `PodPhase` from `public/module/k8s/pods.ts`.